app/components/edit/supplemental_files_step_component.html.erb

```diff
@@ -0,0 +1,25 @@
+<%= render Edit::StepComponent.new(step: SubmissionPresenter::SUPPLEMENTAL_FILES_STEP,
+                                   title: t('steps.supplemental_files.title'),
+                                   data: { controller: 'supplemental_files' },
+                                   submission:,
+                                   done_disabled: done_disabled?) do |component| %>
+  <% component.with_help_content do %>
+    <%= t('steps.supplemental_files.help_html') %>
+  <% end %>
+
+  <% component.with_body_content do %>
+    <%= render Shared::SupplementalFilesStepBodyHelpComponent.new %>
+      <% if supplemental_files.attached? %>
+        <%= helpers.with_submission_form(submission) do |form| %>
+          <%= render Shared::SupplementalFilesStepBodyTableComponent.new(submission:, with_remove: true) %>
+        <% end %>
+      <% end %>
+      <%= helpers.with_submission_form(submission) do |form| %>
```

Comment:
The form must be duplicated here as including the hidden file ids (per: https://guides.rubyonrails.org/active_storage_overview.html#replacing-vs-adding-attachments) causes them to be deleted if there are also hidden ids that mean delete. Duplicating the form solves this.